cypress/components/popover-container/popover-container.cy.tsx

```diff
@@ -338,6 +358,18 @@ context("Test for Popover Container component", () => {
       }
     );
 
+    it("should call onClose callback when the escape is pressed and component is controlled", () => {
+      const callback: PopoverContainerProps["onClose"] = cy
+        .stub()
+        .as("onClose");
+      CypressMountWithProviders(
+        <PopoverContainerComponent onClose={callback} open />
+      );
+
+      popoverCloseIcon().trigger("keydown", keyCode("Esc"));
```

Comment:
```suggestion
      pressESCKey();
```

as in an earlier comment, I think this is a  better test than triggering on the close icon